### ISSUE TITLE
cloned VM but.. Waiting for SSH to become available...

### ISSUE THREAD

**Issue by bala-janakiraman** [nsidc/vagrant-vsphere#157]
Hello,

when I `vagrant up --provider=vsphere` the command line is stuck at `Waiting for SSH to become available...`  and never completes. when i login to vpshere i do see the vm is successfully cloned and i am able to use it through vpshere client.

my vagrant setting is ..
```
Vagrant.configure("2") do |config|
  config.vm.box = 'dummy'
  config.vm.box_url = './example_box/dummy.box'

  config.vm.provider :vsphere do |vsphere|
    vsphere.host = 'ip address1'
    vsphere.compute_resource_name = 'ip address2'
    vsphere.template_name = 'centos7_template'
    vsphere.name = 'vagrant-vsphere'
    vsphere.user = 'xyz'
    vsphere.password = 'abc'
    vsphere.insecure = true
  end
end
``` 

I used `vsphere.insecure = true` to deal with the below error when i do vagrant up.
`SSL_connect returned=1 errno=0 state=SSLv3 read server certificate B: certificate verify failed`. not sure if this is the right way to do it. any help is greatly appreciated.

Bala

**Comment by dbblackdiamond** [comment]
Hi,

when you created your centos7_template, did you download the vagrant.pub key, put it into the authorized_keys file and configured sshd accordingly? 

I had that issue when I first started using vagrant-vsphere and this fixed it.

Thanks a lot,
Bertrand.

**Comment by gaga-github** [comment]
@dbblackdiamond I ran the following on my template and I get the same result as @bala-janakiraman 

mkdir -p /home/vagrant/.ssh
chmod 0700 /home/vagrant/.ssh
wget --no-check-certificate \
    https://raw.github.com/mitchellh/vagrant/master/keys/vagrant.pub \
    -O /home/vagrant/.ssh/authorized_keys
chmod 0600 /home/vagrant/.ssh/authorized_keys
chown -R vagrant /home/vagrant/.ssh

**Comment by dbblackdiamond** [comment]
@gaga-github : did you change the settings in your /etc/ssh/sshd_config file to use the authorized_keys file?

**Comment by bala-janakiraman** [comment]
@dbblackdiamond .. thanks for your reply...

After you pointing out about the pub keys, i copied the key to the authorized_keys file. I am not clear about what you mean by "configure sshd accordingly". could you elaborate please?

on my part.. this is what i did...
1) copied my public key to the authorized_keys file.
2) changed the permission on the ssh folder and the authorized_keys per hashicorp vagrant sites documentation ... http://docs.vagrantup.com/v2/boxes/base.html.. under "Vagrant user" section(essentially what @gaga-github has done).

after this, i created a new VM template, updated my vagrantfile to use the new template, `vagrant up ..`. again stuck at `waiting for ssh to ....`

From my local machine i am able to connect using `ssh -i PRIVATE_KEY user@ipaddress` so the keys work, but not `vagrant ssh`

looking at the sshd_config file .. i presume this is the section of interest for us..

```
# Authentication:

#LoginGraceTime 2m
#PermitRootLogin yes
#StrictModes yes
#MaxAuthTries 6
#MaxSessions 10

#RSAAuthentication yes
#PubkeyAuthentication yes

# The default is to check both .ssh/authorized_keys and .ssh/authorized_keys2
# but this is overridden so installations will only check .ssh/authorized_keys
AuthorizedKeysFile      .ssh/authorized_keys

#AuthorizedPrincipalsFile none

#AuthorizedKeysCommand none
#AuthorizedKeysCommandUser nobody   
```

and  `AuthorizedKeysFile      .ssh/authorized_keys` is the config you are referring to? 

Again i have no idea about configuring sshd.. please enlighten us :-)

**Comment by dbblackdiamond** [comment]
@bala-janakiraman  you are correct that you need to update the /etc/ssh/sshd_config. Here is what you need to change:
```
PubKeyAuthentication yes
AuthorizedKeysFile %h/.ssh/authorized_keys
PermitEmptyPasswords no
```

Once you've changed those lines, you'll need to restart the ssh service, which should be ```service ssh restart```.

**Comment by dbblackdiamond** [comment]
@bala-janakiraman  Did it work for you?

**Comment by gaga-github** [comment]
@bala-janakiraman my problem is with the key
from debug:

    DEBUG ssh: Checking key permissions: /Users/home/.vagrant.d/insecure_private_key
    INFO ssh: Attempting SSH connection...
    INFO ssh: Attempting to connect to SSH...
    INFO ssh:   - Host: 10.251.17.27
    INFO ssh:   - Port: 22
    INFO ssh:   - Username: vagrant
    INFO ssh:   - Password? false
    INFO ssh:   - Key Path: ["/Users/home/.vagrant.d/insecure_private_key"]
    E, [2015-12-30T09:54:12.873184 #4418] ERROR -- net.ssh.authentication.session[82c3db24]: all authorization methods failed (tried none, publickey)

When I try manually, I get the same error, is this the correct key to use: https://raw.github.com/mitchellh/vagrant/master/keys/vagrant.pub on the vm?

-i /Users/home/.vagrant.d/insecure_private_key vagrant@10.251.17.27

    debug1: Trying private key: /Users/home/.vagrant.d/insecure_private_key
    debug1: read PEM private key done: type RSA
    debug1: Authentications that can continue: publickey,gssapi-keyex,gssapi-with-mic,password
    debug1: Next authentication method: password
    vagrant@10.251.17.27's password:

**Comment by bala-janakiraman** [comment]
@dbblackdiamond and @gaga-github , I haven't revisited this issue yet, i will test this out this week hopefully. Appreciate checking in!
